parser/lexer.go

```diff
@@ -628,7 +628,11 @@ func startWithAt(s *Scanner) (tok int, pos Pos, lit string) {
 		tok, lit = scanIdentifierOrString(s)
 		switch tok {
 		case stringLit, quotedIdentifier:
-			tok, lit = doubleAtIdentifier, "@@"+prefix+lit
+			var sb strings.Builder
```

Comment:
What about using append(buf, "@@", prefix, lit) directly? I'm afraid it would need to reallocate the buf a couple of times.

following is the code of strings.Builder.WriteString.
```go
// WriteString appends the contents of s to b's buffer.
// It returns the length of s and a nil error.
func (b *Builder) WriteString(s string) (int, error) {
	b.copyCheck()
	b.buf = append(b.buf, s...)
	return len(s), nil
}
```

Reply:
Yes, you can inline `WriteString` function to get even better performance, and perallocate the buf to avoid slice growth.
But as ChatGPT said:

> Remember that optimizing for performance often comes at the cost of readability and maintainability, so ensure that you balance these concerns when optimizing the code.

I think `StringBuilder` is a common patten and is well known by most of the programmers. So what ChatGPT suggested is indeed the best balanced choice here. @winkyao